src/main/java/com/diffplug/gradle/spotless/FormatTask.java

```diff
@@ -23,19 +23,36 @@
 import java.util.List;
 import java.util.Locale;
 
+import com.diffplug.common.base.Errors;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.GradleException;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.OutputFiles;
+import org.gradle.api.tasks.SkipWhenEmpty;
 import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.incremental.IncrementalTaskInputs;
 
 public class FormatTask extends DefaultTask {
 	// set by SpotlessExtension, but possibly overridden by FormatExtension
+	@Input
 	public Charset encoding = StandardCharsets.UTF_8;
-	public LineEnding.Policy lineEndingsPolicy = LineEnding.UNIX_POLICY;
+	@Input
+	public LineEnding.Policy lineEndingPolicy = LineEnding.UNIX_POLICY;
+
 	// set by FormatExtension
+	@Input
 	public boolean paddedCell = false;
+	@InputFiles
+	@OutputFiles
+	@SkipWhenEmpty
 	public Iterable<File> target;
+	@Input
```

Comment:
> And imo, that's the only place this optimization makes sense anyway.

This is only true in very small projects. In a large project, when you change a single module, you want the other modules to be up-to-date, not spend time re-executing the formatter.